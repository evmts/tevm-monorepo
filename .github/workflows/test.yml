on:
  push:
    branches:
      - main
  pull_request:
  merge_group:

name: Test

jobs:
  test:
    name: Test
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0
          submodules: recursive

      - name: "Setup"
        uses: ./.github/actions/setup

      - name: Run tests
        run: bun nx run-many --target=test:coverage
        env:
          NX_CLOUD_ACCESS_TOKEN: ${{ secrets.NX_CLOUD_ACCESS_TOKEN }}
          RPC_URL_MAINNET: ${{ secrets.RPC_URL_MAINNET }}

      - name: 'Report @evmts/core Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/core"
          working-directory: "./core"
      - name: 'Report @evmts/ethers Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/ethers"
          working-directory: "./ethers"
      - name: 'Report @evmts/bundler Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/bundler"
          working-directory: "./bundlers/bundler"
      - name: 'Report @evmts/bun-plugin Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/bun-plugin"
          working-directory: "./bundlers/bun"
      - name: 'Report @evmts/esbuild-plugin Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/esbuild-plugin"
          working-directory: "./bundlers/esbuild"
      - name: 'Report @evmts/ts-plugin Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/ts-plugin"
          working-directory: "./ts-plugin"
      - name: 'Report @evmts/config Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/config"
          working-directory: "./config"
      - name: 'Report @evmts/esbuild Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/esbuild"
          working-directory: "./examples/esbuild"
  e2e:
    name: E2E
    timeout-minutes: 60
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0
          submodules: recursive

      - name: "Setup"
        uses: ./.github/actions/setup

      - name: Install Playwright Browsers
        run: npx playwright install --with-deps

      - name: Run tests
        run: echo "WIP" # bun nx e2e e2e
        env:
          NX_CLOUD_ACCESS_TOKEN: ${{ secrets.NX_CLOUD_ACCESS_TOKEN }}
