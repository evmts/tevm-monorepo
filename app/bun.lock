{
  "lockfileVersion": 1,
  "workspaces": {
    "": {
      "name": "app",
      "dependencies": {
        "electrobun": "^0.0.18",
      },
      "devDependencies": {
        "@types/bun": "latest",
      },
      "peerDependencies": {
        "typescript": "^5",
      },
    },
  },
  "packages": {
    "@oneidentity/zstd-js": ["@oneidentity/zstd-js@1.0.3", "", { "dependencies": { "@types/emscripten": "^1.39.4" } }, "sha512-Jm6sawqxLzBrjC4sg2BeXToa33yPzUmq20CKsehKY2++D/gHb/oSwVjNgT+RH4vys+r8FynrgcNzGwhZWMLzfQ=="],

    "@types/bun": ["@types/bun@1.2.13", "", { "dependencies": { "bun-types": "1.2.13" } }, "sha512-u6vXep/i9VBxoJl3GjZsl/BFIsvML8DfVDO0RYLEwtSZSp981kEO1V5NwRcO1CPJ7AmvpbnDCiMKo3JvbDEjAg=="],

    "@types/emscripten": ["@types/emscripten@1.40.1", "", {}, "sha512-sr53lnYkQNhjHNN0oJDdUm5564biioI5DuOpycufDVK7D3y+GR3oUswe2rlwY1nPNyusHbrJ9WoTyIHl4/Bpwg=="],

    "@types/filesystem": ["@types/filesystem@0.0.36", "", { "dependencies": { "@types/filewriter": "*" } }, "sha512-vPDXOZuannb9FZdxgHnqSwAG/jvdGM8Wq+6N4D/d80z+D4HWH+bItqsZaVRQykAn6WEVeEkLm2oQigyHtgb0RA=="],

    "@types/filewriter": ["@types/filewriter@0.0.33", "", {}, "sha512-xFU8ZXTw4gd358lb2jw25nxY9QAgqn2+bKKjKOYfNCzN4DKCFetK7sPtrlpg66Ywe3vWY9FNxprZawAh9wfJ3g=="],

    "@types/har-format": ["@types/har-format@1.2.16", "", {}, "sha512-fluxdy7ryD3MV6h8pTfTYpy/xQzCFC7m89nOH9y94cNqJ1mDIDPut7MnRHI3F6qRmh/cT2fUjG1MLdCNb4hE9A=="],

    "@types/node": ["@types/node@22.15.18", "", { "dependencies": { "undici-types": "~6.21.0" } }, "sha512-v1DKRfUdyW+jJhZNEI1PYy29S2YRxMV5AOO/x/SjKmW0acCIOqmbj6Haf9eHAhsPmrhlHSxEhv/1WszcLWV4cg=="],

    "@types/webextension-polyfill": ["@types/webextension-polyfill@0.12.3", "", {}, "sha512-F58aDVSeN/MjUGazXo/cPsmR76EvqQhQ1v4x23hFjUX0cfAJYE+JBWwiOGW36/VJGGxoH74sVlRIF3z7SJCKyg=="],

    "browser-namespace": ["browser-namespace@1.4.0", "", { "dependencies": { "@types/filesystem": "*", "@types/har-format": "*", "@types/webextension-polyfill": "*" } }, "sha512-9b4yNTNs+8HVPssSq8RSZMRunf+G4cVQ2PMtOTn+uEVFOW5C0Uo+eGXuJ5LfxS1UDph5oAdWj92thPyxVhpqXg=="],

    "bun-types": ["bun-types@1.2.13", "", { "dependencies": { "@types/node": "*" } }, "sha512-rRjA1T6n7wto4gxhAO/ErZEtOXyEZEmnIHQfl0Dt1QQSB4QV0iP6BZ9/YB5fZaHFQ2dwHFrmPaRQ9GGMX01k9Q=="],

    "chownr": ["chownr@2.0.0", "", {}, "sha512-bIomtDF5KGpdogkLd9VspvFzk9KfpyyGlS8YFVZl7TGPBHL5snIOnxeshwVgPteQ9b4Eydl+pVbIyE1DcvCWgQ=="],

    "electrobun": ["electrobun@0.0.18", "", { "dependencies": { "@oneidentity/zstd-js": "^1.0.3", "rpc-anywhere": "1.5.0", "tar": "^6.2.1" }, "bin": { "electrobun": "dist/electrobun" } }, "sha512-RyMNGcAaHklicZlJToGfN3fVZGKHpxZv6o8S96TK9tHSY/SRze5bNPIGUnY9wr/BbWuQW5gGUGaVIWWqa5NSZQ=="],

    "fs-minipass": ["fs-minipass@2.1.0", "", { "dependencies": { "minipass": "^3.0.0" } }, "sha512-V/JgOLFCS+R6Vcq0slCuaeWEdNC3ouDlJMNIsacH2VtALiu9mV4LPrHc5cDl8k5aw6J8jwgWWpiTo5RYhmIzvg=="],

    "minipass": ["minipass@5.0.0", "", {}, "sha512-3FnjYuehv9k6ovOEbyOswadCDPX1piCfhV8ncmYtHOjuPwylVWsghTLo7rabjC3Rx5xD4HDx8Wm1xnMF7S5qFQ=="],

    "minizlib": ["minizlib@2.1.2", "", { "dependencies": { "minipass": "^3.0.0", "yallist": "^4.0.0" } }, "sha512-bAxsR8BVfj60DWXHE3u30oHzfl4G7khkSuPW+qvpd7jFRHm7dLxOjUk1EHACJ/hxLY8phGJ0YhYHZo7jil7Qdg=="],

    "mkdirp": ["mkdirp@1.0.4", "", { "bin": { "mkdirp": "bin/cmd.js" } }, "sha512-vVqVZQyf3WLx2Shd0qJ9xuvqgAyKPLAiqITEtqW0oIUjzo3PePDd6fW9iFz30ef7Ysp/oiWqbhszeGWW2T6Gzw=="],

    "rpc-anywhere": ["rpc-anywhere@1.5.0", "", { "dependencies": { "browser-namespace": "^1.4.0" } }, "sha512-ZYrB0foAM4oE7oBnUH3BL7LwtW9d6+RkzL/rFnjj8GCaFt5c81Rbw6oVl6u9AMsGONsKeJX0mL62TpbPXSO6og=="],

    "tar": ["tar@6.2.1", "", { "dependencies": { "chownr": "^2.0.0", "fs-minipass": "^2.0.0", "minipass": "^5.0.0", "minizlib": "^2.1.1", "mkdirp": "^1.0.3", "yallist": "^4.0.0" } }, "sha512-DZ4yORTwrbTj/7MZYq2w+/ZFdI6OZ/f9SFHR+71gIVUZhOQPHzVCLpvRnPgyaMpfWxxk/4ONva3GQSyNIKRv6A=="],

    "typescript": ["typescript@5.8.3", "", { "bin": { "tsc": "bin/tsc", "tsserver": "bin/tsserver" } }, "sha512-p1diW6TqL9L07nNxvRMM7hMMw4c5XOo/1ibL4aAIGmSAt9slTE1Xgw5KWuof2uTOvCg9BY7ZRi+GaF+7sfgPeQ=="],

    "undici-types": ["undici-types@6.21.0", "", {}, "sha512-iwDZqg0QAGrg9Rav5H4n0M64c3mkR59cJ6wQp+7C4nI0gsmExaedaYLNO44eT4AtBBwjbTiGPMlt2Md0T9H9JQ=="],

    "yallist": ["yallist@4.0.0", "", {}, "sha512-3wdGidZyq5PB084XLES5TpOSRA3wjXAlIWMhum2kRcv/41Sn2emQ0dycQW4uZXLejwKvg6EsvbdlVL+FYEct7A=="],

    "fs-minipass/minipass": ["minipass@3.3.6", "", { "dependencies": { "yallist": "^4.0.0" } }, "sha512-DxiNidxSEK+tHG6zOIklvNOwm3hvCrbUrdtzY74U6HKTJxvIDfOUL5W5P2Ghd3DTkhhKPYGqeNUIh5qcM4YBfw=="],

    "minizlib/minipass": ["minipass@3.3.6", "", { "dependencies": { "yallist": "^4.0.0" } }, "sha512-DxiNidxSEK+tHG6zOIklvNOwm3hvCrbUrdtzY74U6HKTJxvIDfOUL5W5P2Ghd3DTkhhKPYGqeNUIh5qcM4YBfw=="],
  }
}
