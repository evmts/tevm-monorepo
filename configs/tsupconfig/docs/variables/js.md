**@tevm/tsupconfig** ∙ [README](../README.md) ∙ [API](../API.md)

***

[API](../API.md) > js

# Variable: js

> **`const`** **js**: `Options`

## Source

[js.js:3](https://github.com/evmts/tevm-monorepo/blob/main/configs/tsupconfig/src/js.js#L3)

***
Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
