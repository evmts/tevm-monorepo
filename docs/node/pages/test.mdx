---
title: Test Page
---

import { Callout } from "vocs/components";

# Test Page

This is a test page to verify that basic MDX works.

<Callout type="info">
This is a simple callout to test MDX rendering.
</Callout>

## Basic Content

Here's some regular markdown:

- Item 1
- Item 2
- Item 3

```javascript
// Test code block
function hello() {
  console.log("Hello world!");
}
```