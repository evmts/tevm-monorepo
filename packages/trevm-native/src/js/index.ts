import { createLogger } from '@tevm/logger';
import { InternalError } from '@tevm/errors';
import type { StateManager } from '@tevm/state';
import type { Common } from '@tevm/common';
import type { Chain } from '@tevm/blockchain';

// Import native module
import { init_trevm, TrevmWrapper } from '../../dist/index.node';

export interface TrevmEvmOptions {
  common: Common;
  stateManager: StateManager;
  blockchain: Chain;
  customPrecompiles?: any[];
  profiler?: boolean;
  allowUnlimitedContractSize?: boolean;
  loggingLevel?: string;
}

export interface TrevmRunCallOptions {
  caller?: string;
  to?: string;
  data?: string;
  value?: bigint;
  gasLimit?: bigint;
  skipBalance?: boolean;
}

export interface TrevmRunCallResult {
  returnValue: string;
  executionGasUsed: string;
  error?: string;
}

export interface TrevmEvm {
  runCall(opts: TrevmRunCallOptions): Promise<TrevmRunCallResult>;
}

/**
 * Creates a trevm-based EVM instance
 * @param options EVM options
 * @returns A trevm EVM instance
 */
export const createTrevm = async ({
  common,
  stateManager,
  blockchain,
  customPrecompiles,
  profiler,
  allowUnlimitedContractSize,
  loggingLevel,
}: TrevmEvmOptions): Promise<TrevmEvm> => {
  const logger = createLogger({
    name: '@tevm/trevm',
    level: loggingLevel ?? 'warn',
  });
  
  logger.debug({
    allowUnlimitedContractSize,
    profiler,
    customPrecompiles: customPrecompiles?.map(c => c.address.toString()),
  });

  try {
    // Initialize the native trevm module with callbacks to the StateManager
    const nativeTrevm = await init_trevm({
      chainId: common.chainId(),
      allowUnlimitedContractSize: allowUnlimitedContractSize ?? false,
      stateCallbacks: {
        getAccount: async (address: string) => {
          try {
            const account = await stateManager.getAccount(address);
            if (!account) return null;
            
            return {
              nonce: account.nonce,
              balance: account.balance,
              codeHash: account.codeHash ? `0x${Buffer.from(account.codeHash).toString('hex')}` : undefined,
              code: account.code ? `0x${Buffer.from(account.code).toString('hex')}` : undefined,
            };
          } catch (error) {
            logger.error(`Error in getAccount for ${address}:`, error);
            return null;
          }
        },
        putAccount: async (address: string, account: any) => {
          try {
            await stateManager.putAccount(address, {
              nonce: account.nonce,
              balance: account.balance,
              codeHash: account.codeHash ? Buffer.from(account.codeHash.slice(2), 'hex') : undefined,
              code: account.code ? Buffer.from(account.code.slice(2), 'hex') : undefined,
            });
            return true;
          } catch (error) {
            logger.error(`Error in putAccount for ${address}:`, error);
            return false;
          }
        },
        getStorage: async (address: string, key: string) => {
          try {
            const value = await stateManager.getContractStorage(address, key);
            return value ? `0x${Buffer.from(value).toString('hex')}` : '0x0';
          } catch (error) {
            logger.error(`Error in getStorage for ${address} key ${key}:`, error);
            return '0x0';
          }
        },
        putStorage: async (address: string, key: string, value: string) => {
          try {
            await stateManager.putContractStorage(
              address,
              key,
              Buffer.from(value.slice(2), 'hex')
            );
            return true;
          } catch (error) {
            logger.error(`Error in putStorage for ${address} key ${key}:`, error);
            return false;
          }
        },
        getCode: async (address: string) => {
          try {
            const code = await stateManager.getContractCode(address);
            return code ? `0x${Buffer.from(code).toString('hex')}` : '0x';
          } catch (error) {
            logger.error(`Error in getCode for ${address}:`, error);
            return '0x';
          }
        },
        putCode: async (address: string, code: string) => {
          try {
            await stateManager.putContractCode(
              address,
              Buffer.from(code.slice(2), 'hex')
            );
            return true;
          } catch (error) {
            logger.error(`Error in putCode for ${address}:`, error);
            return false;
          }
        },
        getBlockHash: async (blockNumber: number) => {
          try {
            const block = await blockchain.getBlock(BigInt(blockNumber));
            return block ? `0x${Buffer.from(block.header.hash()).toString('hex')}` : null;
          } catch (error) {
            logger.error(`Error in getBlockHash for block ${blockNumber}:`, error);
            return null;
          }
        }
      }
    });

    // Return a JavaScript wrapper around the native trevm that matches the Evm interface
    return {
      async runCall(opts: TrevmRunCallOptions): Promise<TrevmRunCallResult> {
        try {
          return await nativeTrevm.run_call({
            caller: opts.caller,
            to: opts.to,
            data: opts.data,
            value: opts.value,
            gas_limit: opts.gasLimit,
            skip_balance: opts.skipBalance,
          });
        } catch (error) {
          logger.error('Error in runCall:', error);
          throw new InternalError(`Trevm execution error: ${error}`);
        }
      }
    };
  } catch (error) {
    logger.error('Error initializing trevm:', error);
    throw new InternalError(`Failed to initialize trevm: ${error}`);
  }
};