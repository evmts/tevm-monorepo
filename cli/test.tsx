import React from 'react';
import chalk from 'chalk';
import test from 'ava';
import {render} from 'ink-testing-library';
import Index from './source/commands/index.js';

test('greet user', t => {
	const {lastFrame} = render(<Index options={{name: 'Jane'}} />);

	t.is(lastFrame(), `Hello, ${chalk.green('Jane')}`);
});
