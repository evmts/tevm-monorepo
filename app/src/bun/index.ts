import { BrowserWindow } from "electrobun/bun";

const win = new BrowserWindow({
  title: "Hello Electrobun",
  url: "views://ui/index.html",
});
