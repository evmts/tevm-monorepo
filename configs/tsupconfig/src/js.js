import { createTsUpOptions } from './createTsupOptions.js'

export const js = createTsUpOptions({
	target: 'js',
})
