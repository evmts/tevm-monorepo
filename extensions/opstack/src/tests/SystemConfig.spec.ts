import { type L1Client, createL1Client } from '../index.js'
import { beforeEach, describe, expect, it } from 'bun:test'

describe('SystemConfig', () => {
	let client: L1Client

	beforeEach(async () => {
		client = createL1Client()
		await client.ready()
	})

	it('should initialize contract correctly', async () => {
		expect(
			await client.contract(client.op.SystemConfig.read.owner()),
		).toMatchSnapshot()
		expect(
			await client.contract(client.op.SystemConfig.read.scalar()),
		).toMatchSnapshot()
		expect(
			await client.contract(client.op.SystemConfig.read.version()),
		).toMatchSnapshot()
		expect(
			await client.contract(client.op.SystemConfig.read.overhead()),
		).toMatchSnapshot()
		expect(
			await client.contract(
				client.op.SystemConfig.read.optimismMintableERC20Factory(),
			),
		).toMatchSnapshot()
		expect(
			await client.contract(client.op.SystemConfig.read.minimumGasLimit()),
		).toMatchSnapshot()
		expect(
			await client.contract(client.op.SystemConfig.read.l1StandardBridge()),
		).toMatchSnapshot()
		expect(
			await client.contract(client.op.SystemConfig.read.batchInbox()),
		).toMatchSnapshot()
		expect(
			await client.contract(client.op.SystemConfig.read.startBlock()),
		).toMatchSnapshot()
		expect(
			await client.contract(client.op.SystemConfig.read.batcherHash()),
		).toMatchSnapshot()
		expect(
			await client.contract(client.op.SystemConfig.read.optimismPortal()),
		).toMatchSnapshot()
	})

	it('can interact with the contract as the owner to change the l2 gas limit', async () => {
		const newGasLimit = 420420420n
		expect(
			await client.contract({
				createTransaction: true,
				from: client.op.SYSTEM_CONFIG_OWNER,
				...client.op.SystemConfig.write.setGasLimit(newGasLimit),
			}),
		).toEqual({
			createdAddresses: new Set(),
			data: undefined,
			executionGasUsed: 6619n,
			gas: 16770596n,
			logs: [
				{
					address: '0x229047fed2591dbec1eF1118d64F7aF3dB9EB290',
					data: '0x0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000190f1b44',
					topics: [
						'0x1d2b0bda21d56b8bd12d4f94ebacffdfb35f5e226f84b461103bb8beab6353be',
						'0x0000000000000000000000000000000000000000000000000000000000000000',
						'0x0000000000000000000000000000000000000000000000000000000000000002',
					],
				},
			],
			rawData: '0x',
			selfdestruct: new Set(),
		})
		expect(
			await client.contract({
				...client.op.SystemConfig.read.gasLimit(),
			}),
		).toHaveProperty('data', newGasLimit)
	})
})
