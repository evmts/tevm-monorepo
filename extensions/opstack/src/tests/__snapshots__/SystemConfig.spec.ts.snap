// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`SystemConfig should initialize contract correctly 1`] = `
{
  "createdAddresses":   Set {},
  "data": "0x9BA6e03D8B90dE867373Db8cF1A58d2F7F006b3A",
  "executionGasUsed": 2418n,
  "gas": 16774797n,
  "logs": [],
  "rawData": "0x0000000000000000000000009ba6e03d8b90de867373db8cf1a58d2f7f006b3a",
  "selfdestruct":   Set {},
}
`;

exports[`SystemConfig should initialize contract correctly 2`] = `
{
  "createdAddresses":   Set {},
  "data": 684000n,
  "executionGasUsed": 2351n,
  "gas": 16774864n,
  "logs": [],
  "rawData": "0x00000000000000000000000000000000000000000000000000000000000a6fe0",
  "selfdestruct":   Set {},
}
`;

exports[`SystemConfig should initialize contract correctly 3`] = `
{
  "createdAddresses":   Set {},
  "data": "1.12.0",
  "executionGasUsed": 604n,
  "gas": 16776611n,
  "logs": [],
  "rawData": "0x00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000006312e31322e300000000000000000000000000000000000000000000000000000",
  "selfdestruct":   Set {},
}
`;

exports[`SystemConfig should initialize contract correctly 4`] = `
{
  "createdAddresses":   Set {},
  "data": 188n,
  "executionGasUsed": 2330n,
  "gas": 16774885n,
  "logs": [],
  "rawData": "0x00000000000000000000000000000000000000000000000000000000000000bc",
  "selfdestruct":   Set {},
}
`;

exports[`SystemConfig should initialize contract correctly 5`] = `
{
  "createdAddresses":   Set {},
  "data": "0x75505a97BD334E7BD3C476893285569C4136Fa0F",
  "executionGasUsed": 2471n,
  "gas": 16774744n,
  "logs": [],
  "rawData": "0x00000000000000000000000075505a97bd334e7bd3c476893285569c4136fa0f",
  "selfdestruct":   Set {},
}
`;

exports[`SystemConfig should initialize contract correctly 6`] = `
{
  "createdAddresses":   Set {},
  "data": 21000000n,
  "executionGasUsed": 2521n,
  "gas": 16774694n,
  "logs": [],
  "rawData": "0x0000000000000000000000000000000000000000000000000000000001406f40",
  "selfdestruct":   Set {},
}
`;

exports[`SystemConfig should initialize contract correctly 7`] = `
{
  "createdAddresses":   Set {},
  "data": "0x99C9fc46f92E8a1c0deC1b1747d010903E884bE1",
  "executionGasUsed": 2472n,
  "gas": 16774743n,
  "logs": [],
  "rawData": "0x00000000000000000000000099c9fc46f92e8a1c0dec1b1747d010903e884be1",
  "selfdestruct":   Set {},
}
`;

exports[`SystemConfig should initialize contract correctly 8`] = `
{
  "createdAddresses":   Set {},
  "data": "0xff00000000000000000000000000000011155420",
  "executionGasUsed": 2448n,
  "gas": 16774767n,
  "logs": [],
  "rawData": "0x000000000000000000000000ff00000000000000000000000000000011155420",
  "selfdestruct":   Set {},
}
`;

exports[`SystemConfig should initialize contract correctly 9`] = `
{
  "createdAddresses":   Set {},
  "data": 0n,
  "executionGasUsed": 2429n,
  "gas": 16774786n,
  "logs": [],
  "rawData": "0x0000000000000000000000000000000000000000000000000000000000000000",
  "selfdestruct":   Set {},
}
`;

exports[`SystemConfig should initialize contract correctly 10`] = `
{
  "createdAddresses":   Set {},
  "data": "0x0000000000000000000000006887246668a3b87f54deb3b94ba47a6f63f32985",
  "executionGasUsed": 2373n,
  "gas": 16774842n,
  "logs": [],
  "rawData": "0x0000000000000000000000006887246668a3b87f54deb3b94ba47a6f63f32985",
  "selfdestruct":   Set {},
}
`;

exports[`SystemConfig should initialize contract correctly 11`] = `
{
  "createdAddresses":   Set {},
  "data": "0xbEb5Fc579115071764c7423A4f12eDde41f106Ed",
  "executionGasUsed": 2494n,
  "gas": 16774721n,
  "logs": [],
  "rawData": "0x000000000000000000000000beb5fc579115071764c7423a4f12edde41f106ed",
  "selfdestruct":   Set {},
}
`;

exports[`SystemConfig can interact with the contract as the owner to change the l2 gas limit 1`] = `
{
  "createdAddresses":   Set {},
  "data": undefined,
  "executionGasUsed": 6619n,
  "gas": 16770596n,
  "logs": [
    {
      "address": "0x229047fed2591dbec1eF1118d64F7aF3dB9EB290",
      "data": "0x0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000190f1b44",
      "topics": [
        "0x1d2b0bda21d56b8bd12d4f94ebacffdfb35f5e226f84b461103bb8beab6353be",
        "0x0000000000000000000000000000000000000000000000000000000000000000",
        "0x0000000000000000000000000000000000000000000000000000000000000002",
      ],
    },
  ],
  "rawData": "0x",
  "selfdestruct":   Set {},
}
`;
